src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -201,18 +202,36 @@ private static void initializeAndUpgradeSchema(
 
     // initialize/upgrade db schema
     Database database = new Database(cassandra, keyspace);
-    int initialVersion = database.getVersion();
-    if (initialVersion > 3 && initialVersion < 9) {
-      // only applicable after `003_switch_to_uuids.cql`
-      // Migration009 needs to happen before `migration.migrate()` in case it fails and needs re-trying
-      Migration009.migrate(session);
-    }
-    MigrationTask migration = new MigrationTask(database, new MigrationRepository("db/cassandra"));
-    migration.migrate();
-    Migration003.migrate(session);
+    int currentVersion = database.getVersion();
+    MigrationRepository migrationRepo = new MigrationRepository("db/cassandra");
+    if (currentVersion < migrationRepo.getLatestVersion()) {
+      LOG.warn("Starting db migration from {} to {}…", currentVersion, migrationRepo.getLatestVersion());
+
+      if (4 <= currentVersion) {
+        List<String> otherRunningReapers = session.execute("SELECT reaper_instance_host FROM running_reapers").all()
+            .stream()
+            .map((row) -> row.getString("reaper_instance_host"))
+            .filter((reaperInstanceHost) -> !AppContext.REAPER_INSTANCE_ADDRESS.equals(reaperInstanceHost))
+            .collect(Collectors.toList());
+
+        Preconditions.checkState(
+            otherRunningReapers.isEmpty(),
+            "Database migration can not happen with other reaper instances running. Found ",
```

Comment:
nitpick: Consider rewording this log message to remove the "not":

> Unable to migrate database while other reaper instances are running. Found

Reply:
bugger, i missed this.